src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -174,7 +174,7 @@ public void ApplyStandardProperties(RestoreRequest request)
                 request.LockFilePath = ProjectJsonPathUtilities.GetLockFilePath(request.Project.FilePath);
             }
 
-            if (request.Project.RestoreMetadata != null) {
+            if (request.Project.RestoreMetadata?.CacheFilePath == null) {
```

Comment:
@nkolev92 seems you flipped the conditional.  Do you expect RestoreMetadata to ever be null?